From 1b6d992ab6cbc1d7a4465fa1096e7c30a0a1b15d Mon Sep 17 00:00:00 2001
From: Jan Kiszka <jan.kiszka@siemens.com>
Date: Mon, 26 Jul 2021 08:21:39 +0200
Subject: [PATCH 15/15] net: ethernet: icssg-prueth: Restore 100M half duplex
 support

This was dropped by 9575e0e5 and consequently also ignored by 808fcf4d.
However, experiments show that at least 100M half duplex is working
fine, also with latest pru firmware.

More details need to come from TI.

Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
---
 drivers/net/ethernet/ti/icssg_prueth.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/ti/icssg_prueth.c b/drivers/net/ethernet/ti/icssg_prueth.c
index a2c4d7bae6be..66994f8c38b7 100644
--- a/drivers/net/ethernet/ti/icssg_prueth.c
+++ b/drivers/net/ethernet/ti/icssg_prueth.c
@@ -557,8 +557,8 @@ static void emac_change_port_speed_duplex(struct prueth_emac *emac,
 	struct prueth *prueth = emac->prueth;
 	int slice = prueth_emac_slice(emac);
 
-	/* only 100M and 1G and full duplex supported for now */
-	if (!(full_duplex && (speed == SPEED_1000 || speed == SPEED_100)))
+	/* only 100M and 1G, and 1G only with full duplex supported for now */
+	if (!((speed == SPEED_1000 && full_duplex) || speed == SPEED_100))
 		return;
 
 	val = icssg_rgmii_get_speed(prueth->miig_rt, slice);
@@ -1682,7 +1682,6 @@ static int prueth_netdev_init(struct prueth *prueth,
 
 	/* remove unsupported modes */
 	emac->phydev->supported &= ~(PHY_10BT_FEATURES |
-				     SUPPORTED_100baseT_Half |
 				     SUPPORTED_1000baseT_Half |
 				     SUPPORTED_Pause |
 				     SUPPORTED_Asym_Pause);
-- 
2.26.2

